test/src/org/omegat/gui/glossary/GlossarySearcherTest.java

```diff
@@ -0,0 +1,272 @@
+/**************************************************************************
+ OmegaT - Computer Assisted Translation (CAT) tool
+          with fuzzy matching, translation memory, keyword search,
+          glossaries, and translation leveraging into updated projects.
+
+ Copyright (C) 2021 Hiroshi Miura
+               Home page: http://www.omegat.org/
+               Support center: https://omegat.org/support
+
+ This file is part of OmegaT.
+
+ OmegaT is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ OmegaT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ **************************************************************************/
+package org.omegat.gui.glossary;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.Test;
+import org.omegat.core.Core;
+import org.omegat.core.TestCore;
+import org.omegat.core.data.EntryKey;
+import org.omegat.core.data.ExternalTMX;
+import org.omegat.core.data.IProject;
+import org.omegat.core.data.PrepareTMXEntry;
+import org.omegat.core.data.ProjectProperties;
+import org.omegat.core.data.ProjectTMX;
+import org.omegat.core.data.SourceTextEntry;
+import org.omegat.core.data.TMXEntry;
+import org.omegat.core.statistics.StatisticsInfo;
+import org.omegat.tokenizer.ITokenizer;
+import org.omegat.tokenizer.LuceneEnglishTokenizer;
+import org.omegat.tokenizer.LuceneJapaneseTokenizer;
+import org.omegat.util.Language;
+
+/**
+ * @author Hiroshi Miura
+ */
+public class GlossarySearcherTest extends TestCore {
+    @Test
+    public void testGlossarySearcherEnglish() {
+        String sourceText = "source";
+        String translationText = "translation";
+        String commentText = "comment";
+        ITokenizer tok = new LuceneEnglishTokenizer();
+        Language language = new Language("en");
+        setupProject(language);
+        List<GlossaryEntry> entries = new ArrayList<>();
+        entries.add(new GlossaryEntry(sourceText, translationText, commentText, true, "origin"));
```

Comment:
As long as you don't need it to be mutable, you can initialize and populate a list in one statement with `Arrays#asList`.